### PR TITLE
Fix boilerplate check crash if .go file has no comments

### PR DESCRIPTION
**Pull request by eparis**
Clayton pointed out that if he created a file with no /\* in it anywhere
the boilerplate logic would crash like:

```
$ hack/verify-boilerplate.sh
Traceback (most recent call last):
  File "hack/../hooks/boilerplate.py", line 87, in <module>
    sys.exit(main())
  File "hack/../hooks/boilerplate.py", line 83, in main
    if not file_passes(filename, extention, ref, p):
  File "hack/../hooks/boilerplate.py", line 38, in file_passes
    while data[0] != "/*\n":
IndexError: list index out of range
```

That is because we were just stripping everything before the first line
that contained exacly "/*".  If no such line existed it got to the end
and just kept going.

This does something smarter. We use a regex to look for one or more
lines which start // +build followed by a single newline and remove only
those.  This obviously found one place where the package name was above
the license and was being missed by both the old and the new checker.

It also fixed the python spew and just tells you your file fails.

### CODE REVIEW

**Comment by eparis** [comment]
@smarterclayton

**Comment by smarterclayton** [comment]
Thanks - LGTM ish but my Python is rusty

File: hooks/boilerplate.py [hunk 1]
```diff
@@ -25,18 +25,20 @@
...
     try:
         f = open(filename, 'r')
     except:
         return False
 
-    data = f.readlines()
+    data = f.read()
 
     # remove build tags from the top of Go file
     if extention == "go":
```
Comment (lavalamp) [inline hooks/boilerplate.py]: extension?

Comment (lavalamp) [inline hooks/boilerplate.py]: (also python seriously lets you use this variable declared in another scope? that is totally not cool.)

Comment (eparis) [inline hooks/boilerplate.py]: huh?  extention was an argument to the function...

Comment (eparis) [inline hooks/boilerplate.py]: now obviously I should learn to spelll

Comment (lavalamp) [inline hooks/boilerplate.py]: ...and I should learn to read, lol!

**Comment by lavalamp** [comment]
LGTMish but my python isn't so great. (and squish commits?)

**Comment by lavalamp** [comment]
Shippable is a flake. LGTM.

**Pull request merged**